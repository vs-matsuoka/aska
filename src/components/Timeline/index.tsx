import { useEffect } from 'react';

// https://blog.narumium.net/2020/04/06/react%E3%81%A7twitter%E3%81%AE%E3%82%BF%E3%82%A4%E3%83%A0%E3%83%A9%E3%82%A4%E3%83%B3%E3%81%AA%E3%81%A9%E3%82%92%E5%9F%8B%E3%82%81%E8%BE%BC%E3%82%80/
const Timeline = () => {
  useEffect(() => {
    document.querySelector("script[src='https://platform.twitter.com/widgets.js']")?.remove?.();
    const s = document.createElement('script');
    s.setAttribute('src', 'https://platform.twitter.com/widgets.js');
    document.body.appendChild(s);
  }, []);
  return (
    <a
      // eslint-disable-next-line tailwindcss/no-custom-classname
      className="twitter-timeline"
      data-width="320"
      data-height="335"
      data-theme="dark"
      data-chrome="nofooter noborders transparent noheader"
      href="https://twitter.com/otoMDM?ref_src=twsrc%5Etfw"
    >
      Tweets by otoMDM
    </a>
  );
};

export default Timeline;
