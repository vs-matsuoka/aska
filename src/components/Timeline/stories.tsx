import { Story, Meta } from '@storybook/react';
import Timeline from '.';

export default {
  title: 'Timeline',
  component: Timeline
} as Meta;

export const Default: Story = () => <Timeline />;
