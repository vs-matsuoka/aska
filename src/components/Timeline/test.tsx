import { render, screen } from '@testing-library/react';

import Timeline from '.';

describe('<Timeline />', () => {
  it('should render the link', () => {
    const { container } = render(<Timeline />);

    expect(screen.getByRole('link')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
