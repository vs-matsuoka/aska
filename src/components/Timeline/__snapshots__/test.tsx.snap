// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timeline /> should render the link 1`] = `
<a
  class="twitter-timeline"
  data-chrome="nofooter noborders transparent noheader"
  data-height="335"
  data-theme="dark"
  data-width="320"
  href="https://twitter.com/otoMDM?ref_src=twsrc%5Etfw"
>
  Tweets by otoMDM
</a>
`;
